src/Functions/FunctionsConversion.h

```diff
@@ -2734,6 +2743,19 @@ class FunctionCast final : public FunctionCastBase
         else if (!can_apply_accurate_cast)
         {
             FunctionPtr function = FunctionTo<ToDataType>::Type::create();
+
+            /// check if conversion function accepts argument's type
+            try
+            {
+                function->getReturnTypeImpl({from_type});
+            }
+            catch (const Exception &)
```

Comment:
Need to check the exception codes, otherwise, it can be a memory limit exception.